packages/core/src/components/image-overlay/image-overlay-overlay.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Prop,
+} from "@stencil/core";
+import * as focusTrap from "focus-trap";
+
+const margin = 16;
+
+@Component({
+  tag: "dso-image-overlay-overlay",
+  styleUrl: "./image-overlay-overlay.scss",
+  shadow: true,
+})
+export class ImageOverlayOverlay {
+  @Element()
+  host!: HTMLElement;
+
+  img!: HTMLImageElement;
+
+  wrapper!: HTMLDivElement;
+
+  trap: focusTrap.FocusTrap | undefined;
+
+  @Prop()
+  src: string | undefined;
+
+  @Prop()
+  alt: string | undefined;
+
+  @Event()
+  close!: EventEmitter<undefined>;
+
+  onClick: (event: MouseEvent | KeyboardEvent) => void = () => {
+    this.trap?.deactivate();
+    this.trap = undefined;
+  };
+
+  stopPropagation: (event: MouseEvent) => void = (event) => {
+    event.stopPropagation();
+  };
+
+  componentDidRender() {
+    this.resizeImage();
+    this.centerImage();
+  }
+
+  resizeImage() {
+    const height = this.host.clientHeight - 2 * margin;
```

Comment:
Ik zie hier allemaal logica die volgens mij veel beter in JSX kan worden uitgeschreven.

Reply:
Ik heb geen idee wat dat betekent.